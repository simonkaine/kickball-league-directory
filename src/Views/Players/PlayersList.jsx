import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { getPlayers } from '../../Services/players.js';

export default function PlayersList() {
    const [players, setPlayers] = useState([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        getPlayers().then((response) => setPlayers(response))

        setTimeout(() => {
            setLoading(false)
        }, 600)
    }, []);

    if(loading) return <h1 style={{height: '100vh', fontSize: '3em', marginTop: '350px'}}>...Loading players</h1>;

    return (
        <section className='Players'>
            <div className='PlayersList'>
            <h1 style={{fontFamily: 'Century Gothic', fontSize: '4em', margin: '100px 0 0 0'}}>~ Players ~</h1>

                <ul aria-label='players' style={{listStyleType: 'none', height: '100vh', margin: '50px 0 0 0'}}>
                    {players.map((player) => {
                        return (
                            <li key={player.id} style={{padding: '10px 0 10px 0', fontSize: '1.5em'}}>
                                <Link to={`/players/${player.id}`}>
                                    {player.name}
                                </Link>
                            </li>
                        );
                    })}
                </ul>
            </div>
        </section>
    )
}