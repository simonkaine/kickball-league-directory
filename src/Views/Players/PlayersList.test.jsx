import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import PlayersList from "./PlayersList";

it('Should test for a detailed players list view', async () => {
    const { container } = render(
        <MemoryRouter  >
            <PlayersList />
        </MemoryRouter>
    );

    screen.getByText('...Loading players');
    const labelText = await screen.findByRole('list', {name: 'players'});
    expect(labelText).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});