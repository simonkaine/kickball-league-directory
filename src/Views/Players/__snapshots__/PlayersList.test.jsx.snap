// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test for a detailed players list view 1`] = `
<div>
  <div>
    <a
      href="/players/create"
    >
      Add a new player
    </a>
  </div>
  <section
    class="Players"
  >
    <div
      class="PlayersList"
    >
      <h1
        style="font-family: Century Gothic; font-size: 4em; margin: 100px 0px 0px 0px;"
      >
        ~ Players ~
      </h1>
      <ul
        aria-label="players"
        style="list-style-type: none; height: 100vh; margin: 50px 0px 0px 0px;"
      >
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/3"
          >
            David Bowie
          </a>
          <a
            href="/players/3/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player David Bowie"
            type="button"
          >
            Delete This player
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/6"
          >
            Don Lego-hands Jr
          </a>
          <a
            href="/players/6/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player Don Lego-hands Jr"
            type="button"
          >
            Delete This player
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/10"
          >
            Dwayne 'the Rock' Johnson
          </a>
          <a
            href="/players/10/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player Dwayne 'the Rock' Johnson"
            type="button"
          >
            Delete This player
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/2"
          >
            Fred Durst
          </a>
          <a
            href="/players/2/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player Fred Durst"
            type="button"
          >
            Delete This player
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/7"
          >
            Linda Ronstadt
          </a>
          <a
            href="/players/7/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player Linda Ronstadt"
            type="button"
          >
            Delete This player
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/4"
          >
            Patrick Stewart
          </a>
          <a
            href="/players/4/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player Patrick Stewart"
            type="button"
          >
            Delete This player
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/1"
          >
            Simon Kaine
          </a>
          <a
            href="/players/1/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player Simon Kaine"
            type="button"
          >
            Delete This player
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 1.5em;"
        >
          <a
            href="/players/9"
          >
            Thomas Shelby
          </a>
          <a
            href="/players/9/update"
          >
            <button>
              Edit Player
            </button>
          </a>
          <button
            aria-label="delete player Thomas Shelby"
            type="button"
          >
            Delete This player
          </button>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
